Qast/ViewControllers/MapViewController.swift

```diff
@@ -115,41 +121,31 @@ extension MapViewController: LocationManagerDelegate {
         if let currentSoundZone = currentSoundZone {
             self.title = currentSoundZone.id
             notificationManager.displaySoundZoneChangeNotification(currentSoundZone, spoken: true)
-            streamingManager.stop()
-            streamingManager.enqueue(currentSoundZone.streamId)
+            delegate?.mapViewController(self, shouldStartPlaying: currentSoundZone)
         } else {
             self.title = "Not in any SoundZone"
             notificationManager.displaySoundZoneChangeNotification(currentSoundZone, spoken: false)
-            streamingManager.stop()
+            delegate?.mapViewController(self, shouldStopPlaying: currentSoundZone)
         }
     }
-
 }
 
 extension MapViewController: SensorDispatchHandler {
     
+    func receivedGesture(type: GestureType, timestamp: SensorTimestamp) {
+        delegate?.mapViewController(self, receivedGesture: type)
+    }
+    
     func receivedRotation(quaternion: Quaternion, accuracy: QuaternionAccuracy, timestamp: SensorTimestamp) {
         let qMap = Quaternion(ix: 1, iy: 0, iz: 0, r: 0)
         let qResult = quaternion * qMap
-        let yaw: Double = (-qResult.zRotation).toDegrees()
+        let yaw: Double = qResult.zRotation.toDegrees()
         
         guard let userLocation = mapView.userLocation?.location, userLocation.horizontalAccuracy > 0 else {
```

Comment:
I think the root cause of MapViewController adopting so many responsibilities beyond controlling the mapView (like being the `SensorDispatchHandler`) is because we use `userLocation` in most of our calculations, and its most easily accessible in the MapViewController.

Maybe we could use application-level state to broadcast `userLocation` direct from LocationManager to the rest of the app. Then SensorDispatchHandling could move up to a more sensible place like `MainViewController`

Still learning iOS best practices, so I'm just porting what would be web-dev best practice to deal with data needed across the app.

`NotificationsCenter?`